### PR TITLE
Azure Functions Module: Only support Amd64

### PR DESCRIPTION
**Pull request by and-rewsmith**
Migrate functions sample to dotnet 6. Changes:
1. Change base image of functions module sample
2. Stop building functions sample module in build pipeline
3. Stop running functions sample module tests on arm32. Introduce new filter for e2e tests `Amd64Only` for tests only running on amd64.

Test in pipelines:
https://dev.azure.com/msazure/One/_build/results?buildId=54903242&view=results

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [x] concise summary of tests added/modified
	- [x] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

**Comment by and-rewsmith** [comment]
FYA since migrating to dotnet 6 the e2e test is suffering from the existing issue where amqp library in SDK doesn't think it has auth context. This appears to be fixed in a new SDK nuget. Will check on release date and will hold off merging this until we get the new bits. Don't want to introduce flakiness.

**Comment by and-rewsmith** [comment]
Darn, it seems like even with the new SDK nuget, we see some of the same issues. Just saw issues this morning :(.

This will require some more investigation by SDK team and I have communicated this.

**Review by gauravIoTEdge — APPROVED** [review]

File: edge-modules/functions/samples/docker/linux/amd64/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿# syntax=docker/dockerfile:1.4
 
-FROM mcr.microsoft.com/azure-functions/dotnet:3.0
+FROM mcr.microsoft.com/azure-functions/dotnet:4.0
```
Comment (gauravIoTEdge) [inline edge-modules/functions/samples/docker/linux/amd64/Dockerfile:3]: Just wondering if this should be something other than 4.0?

Is the dotnet 6 tag somehow not consistent with versioning information? Nothing you can do, I suppose. It's just quirky.

(I did try to pull mcr.microsoft.com/azure-functions/dotnet:6.0 -- and that doesn't exist)

**Review by gauravIoTEdge — COMMENTED** [review]
Your test pipeline build has expired. 

Probably should remove the internal build link anyway.

File: edge-modules/functions/samples/docker/linux/amd64/Dockerfile (hunk 1 above)
Comment (and-rewsmith) [reply edge-modules/functions/samples/docker/linux/amd64/Dockerfile:3]: 4.0 is dotnet 6, so we are good to go.

**Review by damonbarry — APPROVED** [review]

**Pull request merged**